test_scripts/Defects/4_5/1885_SDL_does_not_cut_off_unknown_RPC_and_or_param_from_PTU.lua

```diff
@@ -0,0 +1,158 @@
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require("user_modules/sequences/actions")
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local variables ]]
+-- set default parameters for 'SendLocation' RPC
+local SendLocationParams = {
+  longitudeDegrees = 1.1,
+  latitudeDegrees = 1.1,
+}
+local unknownAPI = "UnknownAPI"
+local unknownParameter = "unknownParameter"
+
+local gpsResponse = {
+  longitudeDegrees = -180,
+  latitudeDegrees = 90,
+  utcYear = 2100,
+  utcMonth = 12,
+  utcDay = 22,
+  utcHours = 20,
+  utcMinutes = 50,
+  utcSeconds = 50,
+  compassDirection = "NORTH",
+  pdop = 1000,
+  hdop = 1000,
+  vdop = 1000,
+  actual = true,
+  satellites = 31,
+  dimension = "2D",
+  altitude = 10000,
+  heading = 359.99,
+  speed = 500,
+  shifted = true
+}
+
+--[[ Local Functions ]]
+
+--[[ @ptuUpdateFuncRPC: update table with unknown RPC for PTU
+--! @parameters:
+--! tbl - table for update
+--! @return: none
+--]]
+local function ptuUpdateFuncRPC(tbl)
+  local VDgroup = {
+    rpcs = {
+      [unknownAPI] = {
+        hmi_levels = { "NONE", "BACKGROUND", "FULL", "LIMITED" },
+        parameters = { "gps" }
+      },
+      SendLocation = {
+        hmi_levels = { "NONE", "BACKGROUND", "FULL", "LIMITED" }
+      }
+    }
+  }
+  tbl.policy_table.functional_groupings["NewTestCaseGroup1"] = VDgroup
+  tbl.policy_table.app_policies[config.application1.registerAppInterfaceParams.fullAppID].groups =
+    { "Base-4", "NewTestCaseGroup1" }
+end
+
+--[[ @ptuUpdateFuncParams: update table with unknown parameters for PTU
+--! @parameters:
+--! tbl - table for update
+--! @return: none
+--]]
+local function ptuUpdateFuncParams(tbl)
+  local VDgroup = {
+    rpcs = {
+      GetVehicleData = {
+        hmi_levels = { "NONE", "BACKGROUND", "FULL", "LIMITED" },
+        parameters = { "gps", unknownParameter }
+      }
+    }
+  }
+  tbl.policy_table.functional_groupings["NewTestCaseGroup2"] = VDgroup
+  tbl.policy_table.app_policies[config.application1.registerAppInterfaceParams.fullAppID].groups =
+    { "Base-4", "NewTestCaseGroup1", "NewTestCaseGroup2" }
+end
+
+--[[ @SuccessfulProcessingRPC: Successful processing API
+--! @parameters:
+--! RPC - RPC name
+--! params - RPC params for mobile request
+--! interface - interface of RPC on HMI
+--! self - test object
+--! @return: none
+--]]
+local function SuccessfulProcessingRPC(RPC, params, interface)
+  local cid = common.getMobileSession():SendRPC(RPC, params)
+  common.getHMIConnection():ExpectRequest(interface .. "." .. RPC, params)
+  -- :Do(function(_,data)
+  --     self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+  --   end)
+  :Do(function(_, data)
+    if RPC == "GetVehicleData" then
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {gps = gpsResponse})
+    elseif
+      RPC == "SubscribeVehicleData" then
```

Comment:
why do we need this code here? There is no successful processing of SubscribeVehicleData

Reply:
@GetmanetsIrina You can find the updates in the commit 93cd00c